src/nft/components/details/detailsV2/BuyButton.tsx

```diff
@@ -114,35 +126,51 @@ export const BuyButton = ({ asset, dataPage: dataPage }: { asset: GenieAsset; da
   )
 
   const navigateToProfile = useCallback(() => {
+    if (!walletAsset) {
+      return
+    }
+
     resetSellAssets()
-    setSellPageState(ProfilePageStateType.VIEWING)
-    clearCollectionFilters()
+    selectSellAsset(walletAsset)
+    sendAnalyticsEvent(NFTEventName.NFT_SELL_ITEM_ADDED, {
+      collection_address: asset.address,
+      token_id: asset.tokenId,
+      ...trace,
+    })
+    setSellPageState(ProfilePageStateType.LISTING)
     navigate('/nfts/profile')
-  }, [clearCollectionFilters, navigate, resetSellAssets, setSellPageState])
+  }, [asset.address, asset.tokenId, navigate, resetSellAssets, selectSellAsset, setSellPageState, trace, walletAsset])
 
   const oneClickBuyAsset = useCallback(() => {
     if (!account) {
       if (!accountDrawerOpen) {
         toggleWalletDrawer()
       }
-      setConnectingToWallet(true)
-      setTimeout(() => setConnectingToWallet(false), 20000)
+      updateConnectingToWallet(true)
+      setTimeout(() => updateConnectingToWallet(false), 30000)
     } else {
       fetchAndPurchaseSingleAsset(asset)
     }
-  }, [account, accountDrawerOpen, asset, fetchAndPurchaseSingleAsset, toggleWalletDrawer])
+  }, [account, accountDrawerOpen, asset, fetchAndPurchaseSingleAsset, toggleWalletDrawer, updateConnectingToWallet])
 
   useEffect(() => {
-    if (connectingToWallet && account) {
-      setConnectingToWallet(false)
+    if (connectingToWallet && account && activationState.status === ActivationStatus.IDLE) {
+      updateConnectingToWallet(false)
       fetchAndPurchaseSingleAsset(asset)
     }
-  }, [connectingToWallet, account, fetchAndPurchaseSingleAsset, asset])
+  }, [
+    connectingToWallet,
+    account,
+    fetchAndPurchaseSingleAsset,
+    asset,
+    updateConnectingToWallet,
+    activationState.status,
+  ])
 
   if (ownsAsset) {
     return (
-      <ButtonContainer dataPage={Boolean(dataPage)}>
-        <StyledBuyButton shouldHide={false} onClick={navigateToProfile}>
+      <ButtonContainer dataPage={Boolean(isOnDataPage)}>
```

Comment:
nit: `isOnDataPage` is already a boolean

Reply:
it needs to be casted to a boolean because it comes in as `boolean | undefined`